eng/pipelines/common/maui-templates.yml

```diff
@@ -125,64 +122,76 @@ jobs:
-    timeoutInMinutes: 240
-    condition: ${{ parameters.condition}}
-    pool: ${{ RunPlatform }}
-    steps:
-
-    - ${{ each step in parameters.prepareSteps }}:
-      - ${{ each pair in step }}:
-          ${{ pair.key }}: ${{ pair.value }}
-
-    - template: provision.yml
-      parameters:
-        skipXcode: ${{ eq(RunPlatform.testName, 'RunOnAndroid') }}
-        checkoutDirectory: ${{ parameters.checkoutDirectory }}
-
-    - task: DownloadBuildArtifacts@0
-      displayName: 'Download Packages'
-      inputs:
-        artifactName: ${{ parameters.artifactName }}
-        itemPattern: ${{ parameters.artifactItemPattern }}
-        downloadPath: $(System.DefaultWorkingDirectory)/artifacts
-
-    - pwsh: Move-Item -Path artifacts\${{ parameters.artifactName }}\*.nupkg -Destination artifacts -Force
-      displayName: Move the downloaded artifacts
-
-    - pwsh: ./build.ps1 --target=dotnet-local-workloads --verbosity=diagnostic
-      displayName: 'Install .NET (Local Workloads)'
-      retryCountOnTaskFailure: 3
-      workingDirectory: ${{ parameters.checkoutDirectory }}
-      env:
-        DOTNET_TOKEN: $(dotnetbuilds-internal-container-read-token)
-        PRIVATE_BUILD: $(PrivateBuild)
-
-    - script: dotnet tool update Microsoft.DotNet.XHarness.CLI --add-source https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json --version "1.0.0-prerelease*" -g
-      displayName: install xharness
-
-    - task: DotNetCoreCLI@2
-      inputs:
-        projects: ${{ parameters.checkoutDirectory }}/src/TestUtils/src/Microsoft.Maui.IntegrationTests/Microsoft.Maui.IntegrationTests.csproj
-      displayName: Build Microsoft.Maui.IntegrationTests
-
-    - task: DotNetCoreCLI@2
-      inputs:
-        command: test
-        projects: ${{ parameters.checkoutDirectory }}/src/TestUtils/src/Microsoft.Maui.IntegrationTests/bin/Debug/net7.0/Microsoft.Maui.IntegrationTests.dll
-        arguments: --logger "console;verbosity=normal" --filter "Name=${{ RunPlatform.testName }}"
-        publishTestResults: true
-        testRunTitle: ${{ RunPlatform.testName }} template run tests
-      displayName: ${{ RunPlatform.testName }} template run tests
-      continueOnError: true
-
-    - pwsh: |
-        Write-Host "Current job status is: $env:AGENT_JOBSTATUS"
-        if ($env:AGENT_JOBSTATUS -eq "SucceededWithIssues") {
-            Write-Host "##vso[task.complete result=Failed;]DONE"
-        }
-      displayName: Fail if any issues occurred
+- job: run_template
+  workspace:
+    clean: all
+  displayName: 'Run template: '
+  timeoutInMinutes: 240
+  condition: ${{ parameters.condition}}
+  strategy:
+    matrix:
+      ${{ each RunPlatform in parameters.RunPlatforms }}:
+        ${{ RunPlatform.testName }}:
+          TEST_NAME: ${{ RunPlatform.testName }}
+          VM_IMAGE: ${{ RunPlatform.vmImage }}
+          OS_VERSION: Ventura
+          OS_ARCH: x64
+
+  pool: 
+    vmImage: $(VM_IMAGE)
```

Comment:
this is missing - name: $(androidTestsVmPool) so we can say  `Azure Pipelines` or `Xamarin Shared`